### PR TITLE
cookies: Statement is in block if(fp) skip check in Line:1233

### PR DESCRIPTION
**Pull request by pheiduck**
https://github.com/curl/curl/pull/8699#discussion_r851583907
Suggested-by: Jay Satiro <raysatiro@yahoo.com>

### CODE REVIEW

File: lib/cookie.c [hunk 1]
```diff
@@ -1230,7 +1230,7 @@ struct CookieInfo *Curl_cookie_init(struct Curl_easy *data,
      */
     remove_expired(c);
 
-    if(fromfile && fp)
+    if(fromfile)
```
Comment (danielgustafsson) [inline lib/cookie.c:1233]: If `fopen()` fails on line 1197, `fromfile` is set but `fp` isn't and shouldn't be `fclose`d. Thus, we need this check.

Comment (pheiduck) [reply lib/cookie.c:1233]: I see, then this is not important.

Comment (jay) [reply lib/cookie.c:1233]: > Thus, we need this check.

no we don't, it's in the if(fp) block

**Comment by danielgustafsson** [comment]
> no we don't, it's in the if(fp) block

Sorry, I was a bit thick, *now* I see what you mean. Will fix.

**Pull request closed without merging**